src/main/java/com/gmail/woosay333/onlinebookstore/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,68 @@
+package com.gmail.woosay333.onlinebookstore.controller;
+
+import com.gmail.woosay333.onlinebookstore.dto.shopping.cart.CartItemDto;
+import com.gmail.woosay333.onlinebookstore.dto.shopping.cart.CartItemUpdateRequestDto;
+import com.gmail.woosay333.onlinebookstore.dto.shopping.cart.CreateCartItemRequestDto;
+import com.gmail.woosay333.onlinebookstore.dto.shopping.cart.ShoppingCartDto;
+import com.gmail.woosay333.onlinebookstore.service.shopping.cart.ShoppingCartService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/cart")
+@RequiredArgsConstructor
+@Tag(name = "Shopping Cart management.",
+        description = "Endpoints for managing Shopping Carts.")
+public class ShoppingCartController {
+    private final ShoppingCartService shoppingCartService;
+
+    @PreAuthorize("hasRole('USER')")
+    @GetMapping
+    @ResponseStatus(HttpStatus.OK)
+    @Operation(summary = "Get a shopping cart.",
+            description = "Retrieve the user's shopping cart from DB.")
+    public ShoppingCartDto getShoppingCart() {
+        return shoppingCartService.getShoppingCart();
+    }
+
+    @PreAuthorize("hasRole('USER')")
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    @Operation(summary = "Add new item to a cart.",
+            description = "Save a new item to the user's shopping cart through DB.")
+    public ShoppingCartDto addToCart(@RequestBody @Valid CreateCartItemRequestDto request) {
+        return shoppingCartService.addToCart(request);
+    }
+
+    @PreAuthorize("hasRole('USER')")
+    @PutMapping("/cart-items/{cartItemId}")
+    @ResponseStatus(HttpStatus.OK)
+    @Operation(summary = "Update a cart item.",
+            description = "Update an existing cart item in the user's shopping cart by its ID.")
+    public CartItemDto updateCartItem(@PathVariable Long cartItemId,
+                                      @RequestBody @Valid CartItemUpdateRequestDto request) {
```

Comment:
Inconsistency in names, you have CreateCartItemRequestDto and CartItemUpdateRequestDto, please decide where to put word that belong to operation. CreateCartItemRequestDto and UpdateCartItemRequestDto or CartItemUpdateRequestDto and CartItemCreateRequestDto